pkg/client/k8s.go

```diff
@@ -621,3 +622,70 @@ func (k K8sClient) GetJVs(volNames []string, rType util.ReturnType, labelSelecto
 	}
 	return nil, nil, errors.New("invalid return type")
 }
+
+// GetLVMvol returns a list or a map of LVMVolume depending upon rType & options
+func (k K8sClient) GetLVMvol(lVols []string, rType util.ReturnType, labelSelector string, options util.MapOptions) (*lvm.LVMVolumeList, map[string]lvm.LVMVolume, error) {
+	lvs := lvm.LVMVolumeList{}
+	// NOTE: The resource name must be plural and the API-group should be present for getting CRs
+	err := k.K8sCS.Discovery().RESTClient().Get().AbsPath("/apis/local.openebs.io/v1alpha1").
+		Resource("lvmvolumes").Do(context.TODO()).Into(&lvs)
+	if err != nil {
+		return nil, nil, err
+	}
+	var list []lvm.LVMVolume
+	if lVols == nil || len(lVols) == 0 {
+		list = lvs.Items
+	} else {
+		lvsMap := make(map[string]lvm.LVMVolume)
+		for _, lv := range lvs.Items {
+			lvsMap[lv.Name] = lv
+		}
+		for _, name := range lVols {
+			if lv, ok := lvsMap[name]; ok {
+				list = append(list, lv)
+			} else {
+				fmt.Printf("Error from server (NotFound): lvmvolume %s not found\n", name)
+			}
+		}
+	}
+	if rType == util.List {
+		return &lvm.LVMVolumeList{
+			Items: list,
+		}, nil, nil
+	}
+	if rType == util.Map {
+		lvMap := make(map[string]lvm.LVMVolume)
+		switch options.Key {
+		case util.Label:
+			for _, lv := range list {
+				if vol, ok := lv.Labels[options.LabelKey]; ok {
+					lvMap[vol] = lv
+				}
+			}
+			return nil, lvMap, nil
+		case util.Name:
+			for _, lv := range list {
+				lvMap[lv.Name] = lv
+			}
+			return nil, lvMap, nil
+		default:
+			return nil, nil, errors.New("invalid map options")
+		}
+	}
+	return nil, nil, errors.New("invalid return type")
+}
+
+// GetCSIControllerPod returns the CSI controller pod with a specific
+// openebs-component-name label key
+func (k K8sClient) GetCSIControllerPod(name string) (*corev1.Pod, error) {
+	if pods, err := k.K8sCS.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("openebs.io/component-name=%s", name),
+	}); err == nil && len(pods.Items) == 1 {
+		// TODO: Think about Evicted pods
```

Comment:
Can we make use of `pod.Spec.Phase` here to cross check Running?

Reply:
We might also need to consider this `evicted` condition while creating `NamespaceMap`

Reply:
I think this can be done via field-selectors, I'll confirm once.

Reply:
Tested it, I've added `status.phase=Running` for `GetOpenEBSNamespace` which is being used for PVC-describe & `GetOpenEBSNamespaceMap` which isn't being used anywhere. I've not added it in the method which gets the TargetPod objects.

Reply:
Instead of pod can we get sts with the label? The sts will have the latest applied configurations and pods can differ

Reply:
> Instead of pod can we get sts with the label?

Are all OpenEBS CSI Controllers STS always? If yes, then that'd be a one solution for all.